main/src/ca/uwaterloo/flix/language/phase/Simplifier.scala

```diff
@@ -307,10 +307,7 @@ object Simplifier extends Phase[TypedAst.Root, SimplifiedAst.Root] {
       case TypedAst.Expression.FixpointProjectOut(_, _, _, _, _) =>
         throw InternalCompilerException(s"Unexpected expression: $exp0.")
 
-      case TypedAst.Expression.MatchEff(_, _, _, _, _, _) =>
-        throw InternalCompilerException(s"Unexpected expression: $exp0.")
-
-      case TypedAst.Expression.IfThenElseStar(_, _, _, _, _, _) =>
+      case TypedAst.Expression.Reify(_, _, _, _) =>
         throw InternalCompilerException(s"Unexpected expression: $exp0.")
```

Comment:
This is because the expression is `@EliminatedBy` `Monomorph`, right?

Reply:
Yes... I wanna say?